apps/extension/src/state/accounts.ts

```diff
@@ -38,21 +41,32 @@ export const createAccountsSlice: SliceCreator<AccountsSlice> = (set, get) => {
         state.accounts.index = index;
       });
     },
-  };
-};
-
-export const accountsSelector = (state: AllSlices) => state.accounts;
-
-export const selectedAccount = (state: AllSlices): Account | undefined => {
-  const active = getActiveWallet(state);
-  if (!active) return undefined;
+    setEphemeral: ephemeral => {
+      set(state => {
+        state.accounts.ephemeral = ephemeral;
+      });
+    },
+    setSelectedAccount: () => {
+      const active = get().wallets.all[0];
+      if (!active) return undefined;
+      const { ephemeral, index } = get().accounts;
 
-  const addr = getAddressByIndex(active.fullViewingKey, state.accounts.index);
-  const bech32Addr = bech32Address(addr);
+      const addr = ephemeral
+        ? getEphemeralByIndex(active.fullViewingKey, index)
+        : getAddressByIndex(active.fullViewingKey, index);
+      const bech32Addr = bech32Address(addr);
 
-  return {
-    address: bech32Addr,
-    preview: getShortAddressByIndex(active.fullViewingKey, state.accounts.index),
-    index: state.accounts.index,
+      set(state => {
+        state.accounts.selectedAccount = {
+          address: bech32Addr,
+          preview: ephemeral
+            ? bech32Addr.slice(0, 33) + '…'
+            : getShortAddressByIndex(active.fullViewingKey, index),
+          index,
+        };
+      });
+    },
```

Comment:
Move this into a selector function instead of a setter like this

Reply:
because of this, there was a problem that after processing the block, the ephemeral address changed

Reply:
Why does do new blocks re-render this component? Regardless, I think that means we need to re-architect this component so re-renders do not influence it like that.

Reply:
Think a React.memo or useMemo would help with this re-render issue